packages/shared/trade/src/maturity-select/components/maturities.tsx

```diff
@@ -34,7 +39,7 @@ export function Maturities({
               }
               onSelect={(key) => {
                 const selected =
-                  key && key !== selectedfCashId ? key : undefined;
+                  key && key !== selectedfCashId ? key : selectedfCashId;
```

Comment:
If key === selectedfCashId just don't call `onSelect` instead.

Reply:
The selected prop does not have access to the key property. It uses selectedfCashId which is why we want to call onSelect